### ISSUE TITLE
[Blocked] ETD testing all the way through accessioning AND updating it

### ISSUE THREAD

**Issue by ndushay** [sul-dlss/infrastructure-integration-test#43]
### create ETD all the way through accessioning

The create_etd_spec isn't really complete -- it doesn't send the object all the way through common accessioning and test that it made it.

Blocked by etdSubmitWF steps (see https://consul.stanford.edu/display/ETD/ETD+Submission+Workflow), which will be moving to the hydra_etd app:
- submit-marc (see sul-dlss/hydra_etd/issues/346) -- run the code to create the desc metadata  to be sent to ILS and intercept it sending the data to ILS?
- check-marc (see sul-dlss/hydra_etd/issues/347) -- skip this functionality and pretend it's ready?
- catalog-status (see sul-dlss/hydra_etd/issues/348) -- alter the desc metadata AS IF it got processed and add it to the repo object?
- other-metadata (see sul-dlss/hydra_etd/issues/348) -- run the code to create the rest of the metadata and add it to the repo object?

In a pinch, we could add the rest of the metadata files to the object directly.

Then we send the object through common-accessioning, and check for results (e.g. in purl page, embargoed ...)

### RE: updating an existing ETD

(this could be a separate test and/or a separate ticket -- it doesn't NEED to happen on the same ETD that was just created -- however, if it's not, the existing ETD should be in the seed data loaded whenever stage objects are wiped out.)

Note that Cathy also asks:

> is it safe to assume that once an ETD is accessioned I will not have a problem opening it up to version it?

**Comment by caaster** [comment]
There is one more action I do quarterly. I manually run a bulk action to release ETDs that were accessioned throughout the quarter to SW. This used to happen automagically up until a couple of years ago and I don't know why it stopped. I assume I will still be able to do this and that perhaps no test is needed -- but just in case I am noting it here.

**Comment by ndushay** [comment]
@andrewjbtw @jcoyne any ideas on this bulk action?  How it is done and where oh where has the cron job gone?

**Comment by andrewjbtw** [comment]
I believe the bulk action is the "Manage release" action that already exists. But IIRC, Cathy has to run multiple reports and diff them in order to identify the druids for the new ETDs that have not yet been released.

I do not know how the release previously worked as I don't think it has worked since before I started at Stanford.

**Comment by jcoyne** [comment]
I don't believe I'd ever seen that before.

**Comment by ndushay** [comment]
From slack

me:
>  I pushed a branch up for infra-int-tests that attempts to fake it through the etd app, but DOES try to do the right thing for otherMetadata.   I'm not sure how close I am to the real app executing, but running against stage I hit an error for shelving in common accessioning.  

jcoyne:
> I'm not a big fan of that approach.  check-marc, submit-marc and other-metadata are the thing we really need to test, but we've just stepped around that.
> We already have tests that show that the accession workflow is working and adding another one isn't necessary.

me:
> fair enough ... but in a perfect world, it would be useful to know that the ETD app changes aren't impacting accessioning.

**Comment by ndushay** [comment]
branch is `moar-etd-steps`

**Comment by mjgiarlo** [comment]
Challenges that are confounding this from going forward:

1. No actual integration with SearchWorks and cataloging workflows
    * Potential workaround: use "the magic catkey," and [inject it when the initial submission is created](https://github.com/sul-dlss/hydra_etd/pull/786).
1. No network access to workflow service
    * Potential workaround: SSH tunneling? (Need to document in README.)
1. No way to trigger workflow-related jobs that are cron-triggered in the ETD application
    * Potential workaround: Supply static content metadata and rights metadata in the test, and otherwise lean on copypasta from the app.

**Comment by ndushay** [comment]
Is it now possible to mock the folio client such that it receives requests from the ETD app and returns the expected information so the ETD app can finish the `etdSubmitWF` and kick off common accessioning, and we can wait for the ETD to show as fully accessioned in Argo?

Effectively, we would be completing the ETD app steps in the argo workflow grid view;  not sure if doing this would be useful.
